crates/ra_db/src/input.rs

```diff
@@ -83,6 +84,7 @@ pub struct CrateGraph {
 #[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
 pub struct CrateId(pub u32);
 
+#[derive(Debug, Clone, PartialEq, Eq)]
 pub struct CrateName(SmolStr);
 
 impl CrateName {
```

Comment:
Speaking about `CrateName::new` I guess there is no big reason to return `&str` in case of error. From my experience it is conventional to accept the owned variant and return it as `Err()` to let the caller further use it.

Reply:
I don't think it makes a big difference, but you can submit another PR, if you want to.